db/document.go

```diff
@@ -1127,6 +1130,11 @@ func (doc *Document) UnmarshalJSON(data []byte) error {
 }
 
 func (doc *Document) MarshalJSON() (data []byte, err error) {
+	// The calling code in ImportDoc will marshal the document since it takes a body with _sync inline.
+	// Make sure that _sync._attachments is present in the document body since _globalSync._attachments_meta
+	// can not be seen in an inline body.
+	doc.SyncData.AttachmentsPre4dot0 = doc.Attachments()
```

Comment:
How sure are we that the on-demand import case in ImportDoc is the only use case for this function, given that it's also the standard override for marshalling of Document?  Were you able to confirm that there aren't any other places we end up calling json.Marshal() on a *Document (including things like passing a *Document to gocb, etc)?